kafka_consumer/assets/configuration/spec.yaml

```
@@ -0,0 +1,246 @@
+name: Kafka Consumer
+files:
+- name: kafka_consumer.yaml
+  options:
+  - name: init_config
+    description: |
+      WARNING: To avoid blindly collecting offsets and lag for an unbounded number
+      of partitions (as could be the case after enabling monitor_unlisted_consumer_groups
+      or monitor_all_broker_highwatermarks) the check collects metrics for at most 500 partitions.
+
+      DEPRECATION NOTICE: In the early days of Kafka, consumer offsets were stored in Zookeeper.
+      So this check currently supports fetching consumer offsets from both Kafka and Zookeeper.
+      However, Kafka 0.9 (released in 2015) deprecated Zookeeper storage. As a result, we have also
+      deprecated fetching consumer offsets from Zookeeper and at some point in the future that
+      functionality will be removed from this check.
+    options:
+    - name: kafka_timeout
+      description: Customizes the Kafka connection timeout.
+      value:
+        type: integer
+        example: 5
+      default: 5
+    - name: zk_timeout
+      description: |
+        DEPRECATED: Customizes the ZooKeeper connection timeout.
+      value:
+        type: integer
+        example: 5
+      default: 5
+  - name: instances
+    description: |
+      
+    options:
+      - name: kafka_connect_str
+        description: |
+          Kafka endpoints and port to connect to.
+          
+          In a production environment, it's often useful to specify multiple
+          Kafka nodes for a single check instance. This way you
+          only generate a single check process, but if one host goes down,
+          KafkaClient tries contacting the next host.
+          Details: https://github.com/DataDog/dd-agent/issues/2943
+        required: True
+        value:
+          type: array
+          items:
+            type: string
+          example: 
+            - localhost:9092
+            - <KAFKA_BROKER_ENDPOINT>:9092
+      - name: kafka_client_api_version
+        description: |
+          Specify the highest client protocol version supported by all brokers in the cluster.
+          
+          This is a performance optimization. If this is not set, then the check automatically probes
+          the cluster for broker version during the connection bootstrapping process. Explicitly setting
+          this bypasses that probe, saving 3-5 network calls depending on the broker version. Note that
+          probing randomly picks a broker to probe, so in a mixed-version cluster, probing returns a
+          non-deterministic result.
+        value:
+          type: string
+          example: "2.3.0"
```

```suggestion
          example: "2.3.0"
          default: null
```
This option doesn't have a default